packages/core/src/Site/index.js

```diff
@@ -1466,6 +1496,147 @@ class Site {
     }
   }
 
+  /**
+   * Archive the current version of the site.
+   *
+   * @param {string} versionName the name of the version
+   * @param {string} archivePath the path to the folder to store the archives in
+   */
+  async archive(versionName, archivePath) {
+    await this.readSiteConfig();
+
+    // Save version data
+    this.versionData = await this.writeVersionsFile(versionName, archivePath);
+    // Exclude versioned files from archiving.
+    this.ignoreVersionFiles('');
+
+    // Used to get accurate intralinks within the archived site:
+    const archivedBaseUrl = this.siteConfig.baseUrl === ''
+      ? `/${archivePath}`
+      : `${this.siteConfig.baseUrl}/${archivePath}`;
+    this.siteConfig.baseUrl = archivedBaseUrl;
+
+    // Create the .tmp folder for storing intermediate results.
+    fs.emptydirSync(this.tempPath);
+    // Clean the output folder; create it if not exist.
+    fs.emptydirSync(this.outputPath);
+
+    try {
+      await this.buildSiteHelper();
+    } catch (error) {
+      await Site.rejectHandler(error, [this.tempPath, this.outputPath]);
+    }
+  }
+
+  /**
+   * Checks the version files of site + subsites and sets them to be ignored in the site config.
+   *
+   * @returns {*}
+   */
+  ignoreVersionFiles(pathToVersionFromRootDir) {
+    const rootVersionPath = path.join(this.rootPath, pathToVersionFromRootDir, VERSIONS_DATA_NAME);
+
+    // find the version file and ignore all archives of the current site
+    if (fs.pathExistsSync(rootVersionPath)) {
+      const rootVersionsJson = fs.readJSONSync(rootVersionPath);
+
+      rootVersionsJson.versions.forEach((vers) => {
+        const filePath = pathToVersionFromRootDir !== ''
+          ? `${pathToVersionFromRootDir}/${vers.archivePath}/**`
+          : `${vers.archivePath}/**`;
+        this.siteConfig.ignore.push(filePath);
+      });
+    }
+
+    // Find versioned subsites, recursively ignore all version directories inside that
+    const pathToDirWithVersion = path.join(this.rootPath, pathToVersionFromRootDir);
+
+    // Do not transfer the versions file into the archived site
+    this.siteConfig.ignore.push(path.join(pathToVersionFromRootDir, VERSIONS_DATA_NAME));
+
+    const pathsToVersionFiles
+     = walkSync(pathToDirWithVersion, { directories: false, ignore: this.siteConfig.ignore })
+       .filter(x => x.endsWith(VERSIONS_DATA_NAME))
+       .map(x => path.relative(pathToDirWithVersion, x)) // assumes versions files are in the 'root' of site
+       .map(x => path.dirname(x));
+
+    pathsToVersionFiles.forEach(p => this.ignoreVersionFiles(p));
```

Comment:
I believe I did reference this section of baseUrlMap, but because I needed an additional map, and was referencing different paths and everything, I thought the additional complexity of extracting it and abstracting this function for re-use would actually make it less maintainable 😅 But if I'm misunderstanding do let me know!